Comorphisms/CASL2NNF.hs

```diff
@@ -0,0 +1,100 @@
+{-# LANGUAGE MultiParamTypeClasses, TypeSynonymInstances, FlexibleInstances #-}
+{- |
+Module      :  $Header$
+Description :  negation normal form
+Copyright   :  (c) Mihai Codescu, 2016
+License     :  GPLv2 or higher, see LICENSE.txt
+
+Maintainer  :  codescu@iws.cs.uni-magdeburg.de
+Stability   :  provisional
+Portability :  non-portable (imports Logic.Comorphism)
+
+-}
+
+module Comorphisms.CASL2NNF where
+
+import Logic.Logic
+import Logic.Comorphism
+
+import CASL.Logic_CASL
+import CASL.AS_Basic_CASL
+import CASL.Sign
+import CASL.Morphism
+import CASL.Sublogic as SL hiding (bottom)
+
+import Common.Result
+import Common.Id
+import qualified Data.Set as Set
+import Common.AS_Annotation
+import Common.ProofTree
+
+data CASL2NNF = CASL2NNF deriving Show
+
+instance Language CASL2NNF where
+    language_name CASL2NNF = "CASL2NNF"
+
+instance Comorphism CASL2NNF
+               CASL CASL_Sublogics
+               CASLBasicSpec CASLFORMULA SYMB_ITEMS SYMB_MAP_ITEMS
+               CASLSign
+               CASLMor
+               Symbol RawSymbol ProofTree
+               CASL CASL_Sublogics
+               CASLBasicSpec CASLFORMULA SYMB_ITEMS SYMB_MAP_ITEMS
+               CASLSign
+               CASLMor
+               Symbol RawSymbol ProofTree where
+    sourceLogic CASL2NNF = CASL
+    sourceSublogic CASL2NNF = SL.caslTop
+    targetLogic CASL2NNF = CASL
+    mapSublogic CASL2NNF  = Just -- TODO: does the sublogic change?
+    map_theory CASL2NNF = mapTheory
+    map_morphism CASL2NNF = return -- morphisms are mapped identically
+    map_sentence CASL2NNF _ s = return $ negationNormalForm s
+    map_symbol CASL2NNF _ = Set.singleton . id
+    has_model_expansion CASL2NNF = True -- check
```

Comment:
this clearly is true, since the model functor is the identity (this argument could be added as a comment).